### DISCUSSION TITLE
CUE Community Call 2023-01-31

### DISCUSSION THREAD

**Discussion by Lyoness** [cue-lang/cue#2219] [Community Calls]
A belated Happy New Year to all! We are excited to confirm **Tuesday 31 January at 1530 UTC** for our first CUE Community Call of 2023. 

Please reply with any questions/topics you would like to see covered.

### Agenda
* Project updates
* Release updates
* Follow-ups from [Community Call 2022-11-29](https://docs.google.com/presentation/d/1vZnHChGex1zRxtR5cQIQMP19gHzA2xaIjkuAMDnSKr8/edit#slide=id.g15b412f612e_0_178)

* Community Updates
* Spotlight on...TBD. (If you would like to be featured on monthly spotlights, let us know!) 

### Meeting details

https://meet.google.com/eyq-uish-zgw

This session will also be recorded for those unable to join us live.

### CUE Community shared calendar
https://calendar.google.com/calendar/u/0?cid=Y19lNzkxMWQ5OWQ4ZGIyMmU2ZTVjMzhkMTVkNjY2ZTVlNjdiNWE5ODNkZWU4N2JmNTU2NDY3NzI1OGIxYjJhMTFhQGdyb3VwLmNhbGVuZGFyLmdvb2dsZS5jb20

### Recording
https://youtu.be/Fukkq9h62xg

### Slides
https://docs.google.com/presentation/d/1n6tbIv7bHj3WKSnKVrsfVCd49DQFBzhlweXkVzcFo5E/edit?usp=sharing

**Comment by kajogo777** [comment]
Looking forward to it!

We launched a new project with CUE and would love to be featured on the spotlight @Lyoness !
https://github.com/devopzilla/guku-devx

**Reply by Lyoness** [reply]
This looks great @kajogo777 🎉

**Comment by wonderflow** [comment]
We'd like to share some usage of CUE in KubeVela, but the time conflicts to our KubeVela community call( bi-weekly Tuesday 1530 UTC ). We can participate next time if the time is available.

CUE is used as our core part in KubeVela ecosystem, in the following scenarios:

1. **Templating**: Creating abstraction for component and traits for application delivery. e.g. [webservice](https://github.com/kubevela/kubevela/blob/master/vela-templates/definitions/internal/component/webservice.cue), [hpa](https://github.com/kubevela/kubevela/blob/master/vela-templates/definitions/internal/trait/hpa.cue).
2. **Executing**: Run user defined workflow with built-in functions such as "base64 encoding", "k8s resource get/apply/delete", "conditional wait" and "data passing".  e.g. [built-in cue functions](https://kubevela.net/docs/platform-engineers/workflow/cue-actions)
3. **Packaging**: We bundle and delivery all kubevela plugins(addons) in vela application format and define them in CUE. e.g. [keda](https://github.com/kubevela/catalog/blob/master/addons/keda), [trivy-operator](https://github.com/kubevela/catalog/blob/master/addons/trivy-operator), [clickhouse](https://github.com/kubevela/catalog/blob/master/experimental/addons/clickhouse), [backstage](https://github.com/kubevela/catalog/blob/master/experimental/addons/backstage).
4. **Extension**: Extend customized function as dynamic CUE package, this feature can be used as a Go library that we call it [CueX - CUE Extension](https://github.com/kubevela/pkg/tree/main/cue/cuex).

**Reply by wonderflow** [reply]
Questions:

1. When will the CUE 0.5.0 be released, and what's the release cadence?
2. Is it possible to add [this feature](https://github.com/cue-lang/cue/issues/1852) in 0.5.x?  We hope to decode CUE into CUE format string when doing packaging, because we nest CUE as string in YAML to apply to K8s.

ping @Somefive @FogDong

**Reply by myitcv** [reply]
@wonderflow - we'd love to have you on a "spotlight" session! Let's pick this up via DM.

Happy to also cover these two questions briefly here but also on the call:

> * When will the CUE 0.5.0 be released, and what's the release cadence?

We have `v0.5.0-beta.3` imminent. If this is stable, we will create a release candidate (or might even jump straight to the final version). 

Per our [last community call of 2022](https://github.com/cue-lang/cue/discussions/2152), the release of v0.5.0 has been dragging on because of countless issues found as part of the cycle and comprehension work. Whilst unfortunate that this release has been quite so extended, all of the work that has gone into this refactor is critical for upcoming work on:

* required fields
* performance
* disjunctions
* query proposal

Another reason for us finding more bugs is that CUE is being used more widely, but also that our testing process via Unity is getting better.

All that said, it is our desire/plan to shorten the release cycle where possible.

> * Is it possible to add [this feature](https://github.com/cue-lang/cue/issues/1852) in 0.5.x?  We hope to decode CUE into CUE format string when doing packaging, because we nest CUE as string in YAML to apply to K8s.

We are feature complete for v0.5.0, so this would be something we consider for v0.6.0 or later. Let's separately pick up this question of prioritising this work.

**Comment by Lyoness** [comment]
Updated calendar entry with meeting details!

The Google Meet Room: https://meet.google.com/eyq-uish-zgw

**Comment by Lyoness** [comment]
The recording and slides links have been added to the announcement above.
